packages/reverse-proxy-service/src/couchdb/resolver.ts

```diff
@@ -0,0 +1,30 @@
+import { createHmac } from 'crypto';
+import { Request, Response, NextFunction } from 'express';
+import { createProxyMiddleware } from 'http-proxy-middleware';
+import { COUCHDB_HOST, COUCHDB_SECRET } from '../config/env';
+
+const useSecureSSL = process.env.NODE_TLS_REJECT_UNAUTHORIZED !== '0';
+
+export default function resolver ( req: Request, res: Response, next: NextFunction ) {
+  const { uid, role, rhatUUID } = res.locals.user;
+
+  /* Adding additional roles */
+  role.push( uid, rhatUUID, 'op-user' );
+
+  const token = createHmac( 'sha1', COUCHDB_SECRET as string ).update( uid ).digest( 'hex' );
```

Comment:
Use ```sha256/sha512``` is suggested ```sha1``` is old compared with latest standards

Reply:
@riginoommen That's the algorithm used by CouchDB for proxy authentication. It's not something we can change.